src/lib/swapper/swappers/LifiSwapper/LifiSwapper2.ts

```diff
@@ -0,0 +1,125 @@
+import { filterCrossChainEvmBuyAssetsBySellAssetId } from '../utils/filterBuyAssetsBySellAssetId/filterBuyAssetsBySellAssetId'
+import { getTradeQuote } from './getTradeQuote/getTradeQuote'
+import { getLifiChainMap } from './utils/getLifiChainMap'
+import { getUnsignedTx } from './utils/getUnsignedTx/getUnsignedTx'
+
+const executedTrades: Map<string, GetStatusRequest> = new Map()
+const tradeQuoteMetadata: Map<string, Route> = new Map()
+
+let lifiChainMapPromise: Promise<Result<Map<ChainId, ChainKey>, SwapErrorRight>> | undefined
+
+export const lifi: Swapper2 = {
+  getTradeQuote: async (
+    input: GetTradeQuoteInput,
+    assets: Partial<Record<AssetId, Asset>>,
+    sellAssetPriceUsdPrecision: string,
+  ): Promise<Result<TradeQuote2, SwapErrorRight>> => {
+    if (lifiChainMapPromise === undefined) lifiChainMapPromise = getLifiChainMap()
+
+    const maybeLifiChainMap = await lifiChainMapPromise
+
+    if (maybeLifiChainMap.isErr()) return Err(maybeLifiChainMap.unwrapErr())
+
+    const tradeQuoteResult = await getTradeQuote(
+      input as GetEvmTradeQuoteInput,
+      maybeLifiChainMap.unwrap(),
+      assets,
+      sellAssetPriceUsdPrecision,
+    )
+
+    return tradeQuoteResult.map(({ selectedLifiRoute, ...tradeQuote }) => {
+      const { receiveAddress, affiliateBps } = input
+      const id = selectedLifiRoute.id
+
+      // store the lifi quote metadata for transaction building later
+      tradeQuoteMetadata.set(id, selectedLifiRoute)
+
+      return { id, receiveAddress, affiliateBps, ...tradeQuote }
+    })
+  },
+
+  getUnsignedTx: async ({ from, tradeQuote, stepIndex }: GetUnsignedTxArgs): Promise<ETHSignTx> => {
+    const lifiRoute = tradeQuoteMetadata.get(tradeQuote.id)
+
+    if (!lifiRoute) throw Error('missing trade quote metadata')
+
+    const { accountNumber, sellAsset } = tradeQuote.steps[stepIndex]
+
+    const unsignedTx = await getUnsignedTx({
+      lifiStep: lifiRoute.steps[stepIndex],
+      accountNumber,
+      sellAsset,
+      from,
+    })
+
+    // cast required because hdwallet typings return ETHSignTx which doesn't explicitly extends SignTx<C>
+    return unsignedTx
+  },
+
+  executeTrade: async ({ txToExecute, wallet, chainId }: ExecuteTradeArgs) => {
+    const adapterManager = getChainAdapterManager()
+    const adapter = adapterManager.get(chainId) as unknown as ChainAdapter<EvmChainId>
+
+    if (wallet.supportsOfflineSigning()) {
+      const signedTx = await adapter.signTransaction({
+        txToSign: txToExecute as ETHSignTx,
+        wallet,
+      })
+
+      const txid = await adapter.broadcastTransaction(signedTx)
+
+      return txid
+    }
+
+    if (wallet.supportsBroadcast() && adapter.signAndBroadcastTransaction) {
+      const txid = await adapter.signAndBroadcastTransaction?.({
+        txToSign: txToExecute as ETHSignTx,
+        wallet,
+      })
+
+      return txid
+    }
+
+    throw Error('sign and broadcast failed')
+  },
```

Comment:
this is copy-paste from your branch @gomesalexandre , minus monads as errors can be simply thrown here due to try/catch in caller with error toast

Reply:
monads made the code very verbose so I opted to keep monads for trade quotes only